src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -361,16 +367,19 @@ def analyze(self, make_template: Callable[[str], Template]) -> JinjaTracer:
                             block_type,
                             self.idx_raw,
                             block_subtype,
+                            block_idx,
                         )
                     )
                     self.raw_slice_info[self.raw_sliced[-1]] = raw_slice_info
-                    block_idx = len(self.raw_sliced) - 1
+                    slice_idx = len(self.raw_sliced) - 1
```

Comment:
Renamed the existing `block_idx` variable to `slice_idx` because that's the more correct name. This allowed me to create a new `block_idx` variable in this PR, with different meaning.